Sources/CryptoSwift/SHA3.swift

```diff
@@ -176,6 +176,43 @@ public final class SHA3: DigestType {
     }
 
     fileprivate func process(block chunk: ArraySlice<UInt64>, currentHash hh: inout Array<UInt64>) {
+        
+        defer {
```

Comment:
I will tend to agree with this, however I still think it's better to avoid 3 level of nesting if we could.
Also method is to long what do you say about moving for loop in a separate method and then we can call the method from guard return?

Reply:
that would (I'm happy to be proved otherwise) result in less performant runtime.